core/src/telemetry/metrics.rs

```diff
@@ -1,16 +1,20 @@
+#[cfg(feature = "telemetry")]
+use crate::telemetry::{
+    default_resource, metric_temporality_to_selector, prometheus_server::PromServer,
+};
 use crate::{
     abstractions::dbg_panic,
-    telemetry::{
-        default_resource, metric_temporality_to_selector, prometheus_server::PromServer,
-        TelemetryInstance, TELEM_SERVICE_NAME,
-    },
+    telemetry::{TelemetryInstance, TELEM_SERVICE_NAME},
 };
+#[cfg(feature = "telemetry")]
```

Comment:
This is a lot of places to introduce the feature guard. I'd say ideally let's push these down into a submodule which is guarded, and re-export them here behind another guard.